// @fluentui/react-message-bar/lib-commonjs/components/MessageBarGroup/useMessageBarGroupStyles.styles.raw.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    messageBarGroupClassNames: function() {
        return messageBarGroupClassNames;
    },
    useMessageBarGroupStyles_unstable: function() {
        return useMessageBarGroupStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _reacttheme = require("@fluentui/react-theme");
const messageBarGroupClassNames = {
    root: 'fui-MessageBarGroup'
};
/**
 * Styles for the root slot
 */ const useStyles = (0, _react.makeStyles)({
    base: {
        animationFillMode: 'forwards',
        animationDuration: _reacttheme.tokens.durationNormal
    },
    enter: {
        animationName: {
            from: {
                opacity: 0,
                transform: 'translateY(-100%)'
            },
            to: {
                opacity: 1,
                transform: 'translateY(0)'
            }
        }
    },
    exit: {
        animationName: {
            from: {
                opacity: 1
            },
            to: {
                opacity: 0
            }
        }
    }
});
const useMessageBarGroupStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(messageBarGroupClassNames.root, state.root.className);
    state.enterStyles = (0, _react.mergeClasses)(styles.base, styles.enter);
    state.exitStyles = (0, _react.mergeClasses)(styles.base, styles.exit);
    return state;
};
