// @fluentui/react-motion-components-preview/lib-commonjs/components/Blur/Blur.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "Blur", {
    enumerable: true,
    get: function() {
        return Blur;
    }
});
const _reactmotion = require("@fluentui/react-motion");
const _fadeatom = require("../../atoms/fade-atom");
const _bluratom = require("../../atoms/blur-atom");
/**
 * Define a presence motion for blur in/out
 *
 * @param fromRadius - The radius of pixels to blend into the blur. A length string, defaulting to '10px'.
 * @param duration - Time (ms) for the enter transition (blur-in). Defaults to the `durationSlow` value (300 ms).
 * @param easing - Easing curve for the enter transition (blur-in). Defaults to the `curveDecelerateMin` value.
 * @param exitDuration - Time (ms) for the exit transition (blur-out). Defaults to the enter duration.
 * @param exitEasing - Easing curve for the exit transition (blur-out). Defaults to the `curveAccelerateMin` value.
 * @param animateOpacity - Whether to animate the opacity. Defaults to `true`.
 */ const blurPresenceFn = ({ fromRadius = '10px', duration = _reactmotion.motionTokens.durationSlow, easing = _reactmotion.motionTokens.curveDecelerateMin, exitDuration = duration, exitEasing = _reactmotion.motionTokens.curveAccelerateMin, animateOpacity = true })=>{
    const enterAtoms = [
        (0, _bluratom.blurAtom)({
            direction: 'enter',
            duration,
            easing,
            fromRadius
        })
    ];
    const exitAtoms = [
        (0, _bluratom.blurAtom)({
            direction: 'exit',
            duration: exitDuration,
            easing: exitEasing,
            fromRadius
        })
    ];
    // Only add fade atoms if animateOpacity is true.
    if (animateOpacity) {
        enterAtoms.push((0, _fadeatom.fadeAtom)({
            direction: 'enter',
            duration,
            easing
        }));
        exitAtoms.push((0, _fadeatom.fadeAtom)({
            direction: 'exit',
            duration: exitDuration,
            easing: exitEasing
        }));
    }
    return {
        enter: enterAtoms,
        exit: exitAtoms
    };
};
const Blur = (0, _reactmotion.createPresenceComponent)(blurPresenceFn);
